main_test.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"testing"
+
+	toolresults "google.golang.org/api/toolresults/v1beta3"
+)
+
+func TestGetSuccessOfExecution_AllSucceed(t *testing.T) {
+	steps := []*toolresults.Step{
+		{
+			DimensionValue: []*toolresults.StepDimensionValueEntry{{Key: "os", Value: "android"}},
+			Outcome:        &toolresults.Outcome{Summary: "success"},
+		},
+		{
+			DimensionValue: []*toolresults.StepDimensionValueEntry{{Key: "os", Value: "ios"}},
+			Outcome:        &toolresults.Outcome{Summary: "success"},
+		},
+	}
+
+	isSuccess, err := GetSuccessOfExecution(steps)
+
+	expected := true
+	if err != nil {
+		t.Errorf("Expected no errors. Go %s", err)
+	}
+
+	if isSuccess != expected {
+		t.Errorf("Expected success to be %v, got %v", expected, isSuccess)
+	}
```

Comment:
We are using the `github.com/stretchr/testify` to simply the assertions. Once you added it to the project you can simply this with
```
import (
	...
	"github.com/stretchr/testify/require"
)

func TestGetSuccessOfExecution_AllSucceed(t *testing.T) {
	...
	isSuccess, err := GetSuccessOfExecution(steps)
	require.NoError(t, err)
	require.True(t, isSuccess)
}
```
You can also add custom messages to the assertions if you want to.